ui/ts/models/status.ts

```diff
@@ -20,87 +21,34 @@ module Models {
       [storeId: string]: Proto.StoreStatus;
     }
 
-    function _availability(status: Proto.Status): string {
-      if (status.leader_range_count === 0) {
-        return "100%";
-      }
-      return Math.floor(status.available_range_count / status.leader_range_count * 100).toString() + "%";
-    }
-
-    function _replicated(status: Proto.Status): string {
-      if (status.leader_range_count === 0) {
-        return "100%";
-      }
-      return Math.floor(status.replicated_range_count / status.leader_range_count * 100).toString() + "%";
+    export interface NodeStatusResponseSet {
+      d: Proto.NodeStatus[];
     }
 
-    const _datetimeFormatter: D3.Time.TimeFormat = d3.time.format("%Y-%m-%d %H:%M:%S");
-    function _formatDate(nanos: number): string {
-      let datetime: Date = new Date(nanos / 1.0e6);
-      return _datetimeFormatter(datetime);
+    export interface NodeStatusMap {
+      [nodeId: string]: Proto.NodeStatus;
     }
 
     export class Stores {
       public allStatuses: Utils.ReadOnlyProperty<Proto.StoreStatus[]>;
 
-      private _data: Utils.QueryCache<StoreStatusMap> = new Utils.QueryCache((): promise<StoreStatusMap> => {
+      private _data: Utils.QueryCache<Proto.StoreStatus[]> = new Utils.QueryCache((): promise<Proto.StoreStatus[]> => {
         return Utils.Http.Get("/_status/stores/")
           .then((results: StoreStatusResponseSet) => {
-            let data: StoreStatusMap = {};
-            results.d.forEach((status: Models.Proto.StoreStatus) => {
-              let storeId: number = status.desc.store_id;
-              data[storeId] = status;
-            });
-            return data;
+            return results.d;
           });
       });
 
-      constructor() {
-        this.allStatuses = Utils.Computed(this._data.result, (map: StoreStatusMap) => {
-          let keys = Object.keys(map).sort();
-          return keys.map((key: string) => map[key]);
+      private _dataMap: Utils.ReadOnlyProperty<StoreStatusMap> = Utils.Computed(this._data.result, (list: Proto.StoreStatus[]) => {
+        let map: StoreStatusMap = {};
+        _.each(list, (status: Models.Proto.StoreStatus) => {
```

Comment:
I thought lodash _.each would revert to array.forEach when available, but it was actually the other way around; lodash polyfills Array.each with their own each() implementation